x/merkledb/proof_test.go

```diff
@@ -22,105 +22,194 @@ import (
 )
 
 func Test_Proof_Empty(t *testing.T) {
-	proof := &Proof{}
-	err := proof.Verify(context.Background(), ids.Empty, 4, DefaultHasher)
-	require.ErrorIs(t, err, ErrEmptyProof)
+	tests := []struct {
+		name   string
+		proof  *Proof
+		expect error
+	}{
+		{
+			name:   "empty proof",
+			proof:  &Proof{},
+			expect: ErrEmptyProof,
+		},
+		{
+			name:   "empty path",
+			proof:  &Proof{Path: []ProofNode{}},
+			expect: ErrEmptyProof,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			err := tt.proof.Verify(context.Background(), ids.Empty, 4, DefaultHasher)
+			require.ErrorIs(err, tt.expect)
+		})
+	}
 }
 
-func Test_Proof_Simple(t *testing.T) {
-	require := require.New(t)
+func Test_Proof_Exclusion(t *testing.T) {
+	tests := []struct {
+		name   string
+		modify func(*Proof)
+		expect error
+	}{
+		{
+			name:   "happy path",
+			modify: func(p *Proof) {},
+		},
+		{
+			name: "proof value not empty",
+			modify: func(p *Proof) {
+				p.Value = maybe.Some([]byte{})
+			},
+			expect: ErrExclusionProofUnexpectedValue,
+		},
+	}
 
-	db, err := getBasicDB()
-	require.NoError(err)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
 
-	ctx := context.Background()
-	require.NoError(db.PutContext(ctx, []byte{}, []byte{1}))
-	require.NoError(db.PutContext(ctx, []byte{0}, []byte{2}))
+			db, err := getBasicDB()
+			require.NoError(err)
 
-	expectedRoot, err := db.GetMerkleRoot(ctx)
-	require.NoError(err)
+			writeBasicBatch(t, db)
 
-	proof, err := db.GetProof(ctx, []byte{})
-	require.NoError(err)
+			for _, k := range []byte{5, 6, 7, 8} {
+				proof, err := db.GetProof(context.Background(), []byte{k})
+				require.NoError(err)
+				require.NotNil(proof)
 
-	require.NoError(proof.Verify(ctx, expectedRoot, db.tokenSize, db.hasher))
-}
+				tt.modify(proof)
 
-func Test_Proof_Verify_Bad_Data(t *testing.T) {
-	type test struct {
-		name        string
-		malform     func(proof *Proof)
-		expectedErr error
+				err = proof.Verify(context.Background(), db.getMerkleRoot(), db.tokenSize, db.hasher)
+				require.ErrorIs(err, tt.expect)
+			}
+		})
 	}
+}
 
-	tests := []test{
+func Test_Proof_Inclusion(t *testing.T) {
+	tests := []struct {
+		name   string
+		modify func(*Proof)
+		expect error
```

Comment:
nit: use `wantErr` or `want` for this name - this file uses `expect/expectedErr` but it's not idiomatic in go.